File: common/src/main/java/guepardoapps/lucahome/common/services/ITemperatureService.java
package guepardoapps.lucahome.common.services;

import android.support.annotation.NonNull;

import guepardoapps.lucahome.common.classes.Temperature;

@SuppressWarnings({"unused"})
public interface ITemperatureService extends ILucaService<Temperature> {
    String TemperatureDownloadFinishedBroadcast = "guepardoapps.lucahome.common.services.temperature.download.finished";
    String TemperatureDownloadFinishedBundle = "TemperatureDownloadFinishedBundle";

    int NotificationId = 64371930;

    void SetActiveTemperature(@NonNull Temperature activeTemperature);

    Temperature GetActiveTemperature();
}

File: common/src/main/java/guepardoapps/lucahome/common/services/temperature/ITemperatureService.kt
package guepardoapps.lucahome.common.services.temperature

import guepardoapps.lucahome.common.models.temperature.Temperature
import guepardoapps.lucahome.common.services.common.ILucaService
import java.util.*

interface ITemperatureService : ILucaService<Temperature> {
    var onTemperatureService: OnTemperatureService?

    fun get(uuid: UUID): Temperature?
}